train_concept_identification.py

```diff
@@ -41,6 +42,90 @@ def compute_loss(criterion, logits, gold_outputs):
   loss = criterion(flattened_logits, flattened_gold_outputs)
   return loss
 
+
+def compute_fScore(gold_outputs, predicted_outputs):
+    """Computes f_score, precision, recall.
+
+  Args:
+    gold_outputs: Gold outputs, shape (output seq len, batch size)
+    predicted_outputs: Predicted outputs, shape (output seq len, batch size)
+
+  Returns:
+    f_score
+  """
+
+    predicted_outputs_no_padding, gold_outputs_no_padding = teasor_to_list(gold_outputs, predicted_outputs)
+
+    ids_to_concepts_list_gold = list(vocabs.concept_vocab.keys())
+    concepts_as_list_gold = []
+    for sentence in gold_outputs_no_padding:
+        concepts_as_list_gold = [ids_to_concepts_list_gold[id] for id in sentence if
+                                 ids_to_concepts_list_gold[id] != UNK]
```

Comment:
I'm not so sure about this. This could artificially increase the dev score. Maybe the initial gold sequence (before numericalization) should be stored instead & used here.
It could be done in a different branch, but please leave at least a TODO here